### ISSUE TITLE
Centrino Advanced-N 6235 not working

### ISSUE THREAD

**Issue by ActuallyNotGelb** [Alex313031/ChromiumOS#4]
Hi! I've recently switched from Arch Linux to CloudReady (I know that sounds weird, but I really only need to ssh onto a friend's server, and have basic commands like adb and fastboot - all of which, I got working), but your builds seemed better.
Sadly, wifi seems to not work whatsoever. Don't your builds include the non-free firmware? CloudReady and ChromeOS Flex seem to include them just fine.

Another question, not related to wifi:
- Could one disable mitigations and force the installation of the Linux containers?

Thanks, in advance!

**Comment by Alex313031** [comment]
@ActuallyNotGelb Yes it includes non-free firmware, however it doesn't have some wifi firmware that are seperate files from the gentoo tree.
And I'm looking into how to do this in source code, but for right now you can do it, but you have to ENABLE mitigations. You also have to go to chrome://flags and disable crostini use DLC. And disable hyperthreading in chrome://flags as well as modifying grub to disable hyperthreading at the kernel level. The force enable for mitigation is done by changing files in /proc and has to be done after every boot.
It's a long annoying process, and I can provide more info here and a link to a cloudready forum describing what people have done to enable crostini.
Can you try this arnold the bat build and see if it works with your wifi card, as an experiment to see if its something on my end since code in my overlay that enables firmware is based on arnolds overlay on github? > https://chromium.arnoldthebat.co.uk/special//Camd64OS_R91-13904.B-Special.7z

**Comment by ActuallyNotGelb** [comment]
> @ActuallyNotGelb Yes it includes non-free firmware, however it doesn't have some wifi firmware that are seperate files from the gentoo tree.
> And I'm looking into how to do this in source code, but for right now you can do it, but you have to ENABLE mitigations. You also have to go to chrome://flags and disable crostini use DLC. And disable hyperthreading in chrome://flags as well as modifying grub to disable hyperthreading at the kernel level. The force enable for mitigation is done by changing files in /proc and has to be done after every boot.
> It's a long annoying process, and I can provide more info here and a link to a cloudready forum describing what people have done to enable crostini.
> Can you try this arnold the bat build and see if it works with your wifi card, as an experiment to see if its something on my end since code in my overlay that enables firmware is based on arnolds overlay on github? > https://chromium.arnoldthebat.co.uk/special//Camd64OS_R91-13904.B-Special.7z

It works just fine on the ISO you've linked, it seems. (It only complains about the missing Google API keys but I read that works on yours I guess).
Also, do you reckon I can get Crostini to work on an i5-3427U chip?

Mitigation: PTE Inversion; VMX: conditional cache flushes, SMT vulnerable

**Comment by ActuallyNotGelb** [comment]
Update: I got the container to work, by changing some grub configs.

How-to:

Just add 'l1tf=full mds=full' to the local B image, in /boot/efi/boot/grub.cfg (the tutorial also told me to change the timeout to 10 seconds, will investigate)

**Comment by Alex313031** [comment]
@ActuallyNotGelb Ok then that means it is something on my end. And yes, his builds stopped integrating api keys in march 2021.

And yup thats basically it as far as what ive done based on that cloudready post.

**Comment by ActuallyNotGelb** [comment]
Aight. I'm up to test the wifi, if you want (Pretty sure these cards are quite popular anyway). But the fact I can't enable hyperthreading, and the fact that ChromiumOS is overall pretty crude, made me switch back to Arch Linux.

Sorry :/

![image](https://user-images.githubusercontent.com/70820860/155407963-c531ec71-3006-43fb-a9c6-c88530c74b70.png)

**Comment by Alex313031** [comment]
@ActuallyNotGelb I will be working on adding firmware, and making sure it's actually working properly. I've also got a build with preliminary nvidia support. Next build will hopefully be more like cloudready in terms of hardware support.

**Comment by Alex313031** [comment]
@ActuallyNotGelb Try latest release if you wish to see if it fixes it.
https://github.com/Alex313031/ChromiumOS/releases/tag/V102.0.4964.1

**Comment by ActuallyNotGelb** [comment]
> @ActuallyNotGelb Try latest release if you wish to see if it fixes it. https://github.com/Alex313031/ChromiumOS/releases/tag/V102.0.4964.1

Sorry for not reporting anything back, but my laptop had a hardware failure, and I honestly didn't have the time, patience, or money to go and fix it, so I've been running my mum's Toshiba for the past few months, with Linux - and so I can't test Centrino wifi

**Comment by ActuallyNotGelb** [comment]
my bad, didn't mean to close anything bruh
